### PR TITLE
chore: AAVE collateral liquidation (SC-2752)

### PR DESCRIPTION
**Pull request by satyamakgec**
# Description

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

File: contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    function test_collateralOnboardingAave1(
+        uint256 apr,
+        uint256 index,
+        uint256 numPayments,
+        uint256 requestAmount,
+        uint256 collateralRatio,
+        uint256 fundAmount,
+        uint256 drawdownAmount
+    ) public {
+        createLoanAndDrawdown(apr, index, numPayments, requestAmount, collateralRatio, fundAmount, drawdownAmount,"AAVE");
```
Comment (lucas-manuel) [inline contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol]: `drawdownAmount, "AAVE");`

File: contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+    function createLoanAndDrawdown(
+        uint256 apr,
+        uint256 index,
+        uint256 numPayments,
+        uint256 requestAmount,
+        uint256 collateralRatio,
+        uint256 fundAmount,
+        uint256 drawdownAmount,
+        bytes32 collateralTicker
```
Comment (lucas-manuel) [inline contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol]: `collateralSymbol`

File: contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        assertEq(loan.principalOwed(),                      0);  // Principal owed
+        assertEq(uint256(loan.loanState()),                 0);  // Loan state: Ready
+
+        // Fee related variables pre-check.
+        assertEq(loan.feePaid(),                            0);  // feePaid amount
+        assertEq(loan.excessReturned(),                     0);  // excessReturned amount
+        assertEq(usdc.balanceOf(address(treasury)),         0);  // Treasury liquidityAsset balance
+
+        // Pause protocol and attempt drawdown()
+        assertTrue(emergencyAdmin.try_setProtocolPause(address(globals), true));
```
Comment (lucas-manuel) [inline contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol]: No assertion needed: `emergencyAdmin.setProtocolPause(address(globals), true);`

File: contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        assertTrue(!bob.try_drawdown(address(loan), drawdownAmount));
+
+        // Unpause protocol and drawdown()
+        assertTrue(emergencyAdmin.try_setProtocolPause(address(globals), false));
```
Comment (lucas-manuel) [inline contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol]: No assertion needed: `emergencyAdmin.setProtocolPause(address(globals), false);`

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
LGTM, small changes

File: contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        uint256 collateralRatio,
+        uint256 fundAmount,
+        uint256 drawdownAmount
+    ) public {
+        // Assertion for the oracle price check.
+        assertTrue(globals.getLatestPrice(AAVE) > uint256(0));
+
+        createLoanAndDrawdown(apr, index, numPayments, requestAmount, collateralRatio, fundAmount, drawdownAmount, "AAVE");
+    }
+}
```
Comment (deluca-mike) [inline contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol]: Newline

**Review by deluca-mike — CHANGES_REQUESTED** [review]
1 new line needed, and just confirming that the dapp-cache is supposed to be updated in this commit.

File: contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol (hunk 5 above)
Comment (satyamakgec) [reply contracts/core/loan/v1/test/LoanCollateralAssetOnboarding.t.sol]: I haven't updated the dapp-cache from my local as it breaks the test

**Review by deluca-mike — APPROVED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**